includes/admin/settings/class-email-marketing.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Email Marketing
+ *
+ * Manages automatic installation/activation for email marketing extensions.
+ *
+ * @package     EDD
+ * @subpackage  EmailMarketing
+ * @copyright   Copyright (c) 2021, Easy Digital Downloads
+ * @license     https://opensource.org/licenses/gpl-2.0.php GNU Public License
+ * @since       2.11.x
+ */
+namespace EDD\Admin\Settings;
+
+use \EDD\Admin\Extensions\Extension;
+
+class EmailMarketing extends Extension {
+
+	public function __construct() {
+		add_filter( 'edd_settings_sections_marketing', array( $this, 'add_section' ) );
+		add_action( 'edd_settings_tab_top_marketing_email_marketing', array( $this, 'field' ) );
+
+		parent::__construct();
+	}
+
+	/**
+	 * Gets the configuration for the newsletter extensions.
+	 *
+	 * @since 2.11.x
+	 * @return array
+	 */
+	protected function get_configuration( $item_id = false ) {
+		$extensions = include 'list-email-marketing.php';
+
+		return $item_id ? $extensions[ $item_id ] : $extensions;
+	}
+
+	/**
+	 * Adds an email marketing section to the Marketing tab.
+	 *
+	 * @since 2.11.x
+	 * @param array $sections
+	 * @return array
+	 */
+	public function add_section( $sections ) {
+		if ( $this->is_activated() ) {
+			return $sections;
+		}
+
+		$sections['email_marketing'] = __( 'Email Marketing', 'easy-digital-downloads' );
+
+		return $sections;
+	}
+
+	/**
+	 * Adds the email marketing extensions as cards.
+	 *
+	 * @since 2.11.x
+	 * @return void
+	 */
+	public function field() {
+		$config = $this->get_configuration();
+		?>
+		<div class="edd-extension-manager__card-group">
+			<?php
+			foreach ( $config as $item_id => $extension ) {
+				$this->do_single_extension_card( $item_id );
+			}
+			?>
+		</div>
+		<style>p.submit{display:none;}</style>
+		<?php
+	}
+
+	/**
+	 * Overrides the body array sent to the Products API.
+	 *
+	 * @since 2.11.x
+	 * @return array
+	 */
+	protected function get_api_body() {
+		return array(
+			'tag' => 1578,
+		);
+	}
+
+	/**
+	 * Whether any email marketing extension is active.
+	 *
+	 * @since 2.11.x
+	 *
+	 * @return bool True if any email marketing extension is active.
+	 */
+	protected function is_activated() {
+		$config = $this->get_configuration();
+		foreach ( $config as $extension ) {
+			if ( is_plugin_active( $extension['basename'] ) ) {
```

Comment:
I'm putting this here but I want it to apply to all `is_activated()` methods that extend the `Extension` class.

I have some concerns about relying on `is_plugin_active()`. It will work in most cases, but not if someone has renamed the plugin directory. Software Licensing is a great example. (Not in use in this PR, but a good example.)

My SL directory is `EDD-Software-Licensing`, as that's the repo name. But for most of our customers, it will be `edd-software-licensing`. Point being: a check for `is_plugin_active( 'edd-software-licensing/edd-software-licenses.php' )` will _fail_ for me.

Perhaps we can improve this by using multiple checks. If you merge in `release/2.11.4` then I think the `ExtensionRegistry` could help with this. We could add a new method called `isExtensionActive( $extensionId )` (takes the EDD product ID). This could just return `$this->offsetExists( $extensionId )`.

Obviously this will only work for new extensions, but it's a good way forward, and would just be one of several.

Then the general flow for `is_activated()` could be this:

1. Check `is_plugin_active()`. If that returns `true`, return `true`. Otherwise continue to next checks.
1. Check extension registry `isExtensionActive()`.
2. We could potentially use the `$edd_licensed_products` global. That only adds products that have a license key entered, so it's not great, but could be a second step.
3. I noticed that the Recurring one does a `class_exists()` check. We could add something to the configs for a certain function to check.